### PR TITLE
Make certain blocks not replacable and indestructible by raiders

### PR DESCRIPTION
**Pull request by Raycoms**
# Changes proposed in this pull request:
- Fixes raiders able to break bedrock or hut blocks
- Fixes players being able to do so by placing hut blocks into bedrock
-

Review please

### CODE REVIEW

File: src/main/resources/assets/minecolonies/lang/en_us.json [hunk 1]
```diff
@@ -1935,6 +1935,6 @@
   "com.minecolonies.coremod.gui.townhalltab.citizens": "Citizens",
   "com.minecolonies.coremod.gui.townhalltab.settings": "Settings",
   "com.minecolonies.coremod.gui.townhalltab.workorders": "Work Orders",
-  "com.minecolonies.coremod.gui.townhalltab.happiness": "Happiness"
-
+  "com.minecolonies.coremod.gui.townhalltab.happiness": "Happiness",
+  "com.minecolonies.coremod.buildtool.indestructible": "Failed to place hut block, block at pos is indestructible!"
```
Comment (ravenbuilder934) [inline src/main/resources/assets/minecolonies/lang/en_us.json]:
```suggestion
  "com.minecolonies.coremod.buildtool.indestructible": "Failed to place the hut block! The block at its intended position is indestructible!"
```
If this is a chat message you have some more space

File: src/api/java/com/minecolonies/api/entity/pathfinding/PathingStuckHandler.java [hunk 2]
```diff
@@ -356,24 +359,46 @@ private void resetStuckTimers()
...
+     * Check if the block at the position is indestructible, if not, attempt to break it.
+     * @param world the world the block is in.
+     * @param pos the pos the block is at.
+     */
+    private void setAirIfPossible(final World world, final BlockPos pos)
+    {
+        final Block blockAtPos = world.getBlockState(pos).getBlock();
+        if (blockAtPos instanceof IBuilderUndestroyable || ModTags.indestructible.contains(blockAtPos))
+        {
+            world.setBlockState(pos, Blocks.AIR.getDefaultState());
```
Comment (Minecraftschurli) [inline src/api/java/com/minecolonies/api/entity/pathfinding/PathingStuckHandler.java]: shouldn't this be inverted?

Comment (someaddons) [reply src/api/java/com/minecolonies/api/entity/pathfinding/PathingStuckHandler.java]: good catch^^

File: src/main/java/com/minecolonies/coremod/network/messages/server/BuildToolPlaceMessage.java [hunk 3]
```diff
@@ -51,6 +53,7 @@
      */
     private static final String NO_HUT_IN_INVENTORY = "com.minecolonies.coremod.gui.buildtool.nohutininventory";
     private static final String WRONG_COLONY        = "com.minecolonies.coremod.gui.buildtool.wrongcolony";
+    private static final String IMMUTABLE_BLOCK_AT_POS = "com.minecolonies.coremod.buildtool.indestructible";
```
Comment (someaddons) [inline src/main/java/com/minecolonies/coremod/network/messages/server/BuildToolPlaceMessage.java]: indestructible block not immutable

File: src/main/java/com/minecolonies/coremod/network/messages/server/BuildToolPlaceMessage.java [hunk 4]
```diff
@@ -200,6 +203,14 @@ private static void handleHut(
       final boolean mirror,
       final BlockState state)
     {
+        final Block blockAtPos = world.getBlockState(buildPos).getBlock();
+        if (blockAtPos instanceof IBuilderUndestroyable || ModTags.indestructible.contains(blockAtPos))
+        {
+            LanguageHandler.sendPlayerMessage(player, IMMUTABLE_BLOCK_AT_POS);
+            SoundUtils.playErrorSound(player, buildPos);
+            return;
```
Comment (someaddons) [inline src/main/java/com/minecolonies/coremod/network/messages/server/BuildToolPlaceMessage.java:211]: wouldnt this also prevent pasting over an existing building?

File: src/api/java/com/minecolonies/api/entity/pathfinding/PathingStuckHandler.java (hunk 2 above)
Comment (Raycoms) [reply src/api/java/com/minecolonies/api/entity/pathfinding/PathingStuckHandler.java]: indeed

File: src/main/java/com/minecolonies/coremod/network/messages/server/BuildToolPlaceMessage.java (hunk 4 above)
Comment (Raycoms) [reply src/main/java/com/minecolonies/coremod/network/messages/server/BuildToolPlaceMessage.java:211]: No, only placing, pasting is a different message

**Review by someaddons — APPROVED** [review]

**Review by Minecraftschurli — APPROVED** [review]

**Pull request merged**